controller/src/main/java/io/pravega/controller/store/stream/ZKStream.java

```diff
@@ -442,24 +421,42 @@ public void refresh() {
     }
 
     @Override
+    CompletableFuture<Data<Integer>> getHistoryTableFromStore() {
+        cache.invalidateCache(historyPath);
+        return getHistoryTable();
+    }
+
+    @Override
+    CompletableFuture<Void> createEpochNode(int epoch) {
+        return store.createZNodeIfNotExist(getEpochPath(epoch));
+    }
+
+    @Override
+    CompletableFuture<Void> deleteEpochNode(int epoch) {
```

Comment:
Is this operation idempotent? We need to make it idempotent, so as to simplify caller's logic.